mqttrust_core/src/lib.rs

```diff
@@ -15,6 +15,7 @@ pub use mqttrs::{
     Connect, Packet, Protocol, Publish, QoS, QosPid, Suback, Subscribe, SubscribeReturnCodes,
     SubscribeTopic, Unsubscribe,
 };
+pub use mqttrust::{PublishPayload, PublishRequest, Request, SubscribeRequest, UnsubscribeRequest};
```

Comment:
Also needs the trait itself

Reply:
It is intentional. A user can import the trait by saying `mqttrust::Client` and its impl `mqttrust_core::Client`.

Reply:
That's not how trait implementations work.. The `mqttrust_core::Client` has no function without the trait in scope. That effectively means that for `factbird-mini` for example, we would need both `mqttrust` and `mqttrust_core` in the dependency list. This can be avoided by re-exporting the trait through `mqttrust_core`

Reply:
OK, `mqttrust::Client` -> `mqttrus::Mqtt` and a user will `use mqttrust::Mqtt; use mqttrust_core::Client;`.

Reply:
It should be enough to `use mqttrust_core::{Mqtt, Client}`.
No reason to depend on `mqttrust` if you already depend on `mqttrust_core`

Reply:
Now I made 2 PRs (`rustot` and `factbird-mini`). Can you have a look?

Reply:
Sure! Will take a look as soon as i get a minute :+1: